secret.go

```diff
@@ -57,8 +57,23 @@ const (
 //   ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789!@#$%^&*()-
 //
 // RandomPassword does not support Handler (user-invoked password rotation),
-// it only supports rotation by Secrets Manager.
-type RandomPassword struct{}
+// it only supports rotation by Secrets Manager. The password generated by
+// RandomPassword may be configured by setting either `PasswordLength` or
+// `ValidCharset` on initialization
+type RandomPassword struct {
+	// Options to configure the random password generated.
+
+	// PasswordLength defines the length of the password generated by RandomPassword.
+	// If not provided, DEFAULT_PASSWORD_LENGTH will be used
+	PasswordLength *int
```

Comment:
Avoid pointers to primitive in Go. Idiomatic Go just uses zero values. In this case, it's fair to say that a password len cannot be zero chars long, so if not specified (zero value) then the default is used.